spec/bundler/shared_helpers_spec.rb

```diff
@@ -0,0 +1,23 @@
+require "spec_helper"
+
+require "bundler/shared_helpers"
+
+RSpec.describe Bundler::SharedHelpers do
+  describe "#set_bundle_environment" do
+    it "does not raise an exception when the filesystem prsents utf-8 paths and env is windows-1251 encoded" do
+      # cyrillic c encoded with windows-1251
+      allow(ENV).to receive(:[]).and_return([209].pack("c*").force_encoding("windows-1251"))
```

Comment:
lolsob indeed it does not. We'll have to guard with `responds_to?` while we support 1.8.7.